src/runtime/genericutil.cs

```diff
@@ -75,20 +75,27 @@ public static List<string> GetGenericBaseNames(string ns)
         }
 
         /// <summary>
-        /// xxx
+        /// Finds a generic type with the given number of generic parameters and the same name and namespace as <paramref name="t"/>.
         /// </summary>
         public static Type GenericForType(Type t, int paramCount)
         {
             return GenericByName(t.Namespace, t.Name, paramCount);
         }
 
+        /// <summary>
+        /// Finds a generic type in the given namespace with the given name and number of generic parameters.
+        /// </summary>
         public static Type GenericByName(string ns, string name, int paramCount)
         {
-            foreach (Type t in GenericsByName(ns, name))
+            var types = GenericsByName(ns, name);
```

Comment:
Is this function used somewhere else? Maybe it should just not return `null`.

Reply:
It is also used by GenericsForType, but that function seems to be unused itself.  Another option is to return an empty list.  However, that would involve an allocation.  Array.Empty makes more sense, which would involve some more changes.

Reply:
Perhaps the best approach is to inline GenericsByName into GenericByName.